pkg/reconciler/revision/revision.go

```diff
@@ -78,21 +93,76 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-			v1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
-	}
 
-	rev.Status.ImageDigest = digest
+	var digestGrp errgroup.Group
+	type digestData struct {
+		digestValue        string
+		containerName      string
+		isServingContainer bool
+		image              string
+		digestError        error
+	}
 
+	digests := make(chan digestData, len(rev.Spec.Containers))
+	for _, container := range rev.Spec.Containers {
+		container := container // Standard Go concurrency pattern.
+		digestGrp.Go(func() error {
+			digest, err := c.resolver.Resolve(container.Image,
+				opt, cfgs.Deployment.RegistriesSkippingTagResolving)
+			if err != nil {
+				err = fmt.Errorf("failed to resolve image to digest: %w", err)
+				digests <- digestData{
+					image:       container.Image,
+					digestError: err,
+				}
+			} else {
+				digests <- digestData{
+					digestValue:        digest,
+					containerName:      container.Name,
+					isServingContainer: len(rev.Spec.Containers) == 1 || len(container.Ports) != 0,
+				}
+			}
+			return nil
+		})
+	}
+	digestGrp.Wait()
+	close(digests)
+	for v := range digests {
+		if v.digestError != nil {
+			rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
+				v1.RevisionContainerMissingMessage(
+					v.image, v.digestError.Error()))
+			return v.digestError
+		}
+		if v.isServingContainer {
+			rev.Status.DeprecatedImageDigest = v.digestValue
+		}
+		rev.Status.ImageDigests = append(rev.Status.ImageDigests, v1.ImageDigests{
+			ContainerName: v.containerName,
+			ImageDigest:   v.digestValue,
+		})
+	}
+	rev.Status.ImageDigests = removeDuplicateDigests(rev.Status.ImageDigests)
 	return nil
 }
 
+func removeDuplicateDigests(imageDigests []v1.ImageDigests) []v1.ImageDigests {
+	digests := make(map[v1.ImageDigests]bool)
+	list := []v1.ImageDigests{}
+	for _, d := range imageDigests {
+		if v := digests[d]; !v {
+			digests[d] = true
+			list = append(list, d)
+		}
+	}
+	for i, v := range list {
+		if v == (v1.ImageDigests{}) {
```

Comment:
This was happening because of the incorrect `make` call above so you can drop this